test/e2e/discovery/service_discovery.go

```diff
@@ -332,6 +406,54 @@ func verifyServiceIpWithDig(f *framework.Framework, srcCluster, targetCluster fr
 	})
 }
 
+func verifyRoundRobinWithDig(f *framework.Framework, srcCluster framework.ClusterIndex, serviceName string, serviceIPList []string,
+	targetPod *corev1.PodList, domains []string) {
+	cmd := []string{"dig", "+short"}
+
+	for i := range domains {
+		cmd = append(cmd, serviceName+"."+f.Namespace+".svc."+domains[i])
+	}
+
+	op := "is"
+	serviceIPMap := make(map[string]int)
+
+	By(fmt.Sprintf("Executing %q to verify IPs %q for service %q %q discoverable", strings.Join(cmd, " "), serviceIPList, serviceName, op))
+
+	for count := 0; count < 10; count++ {
+		framework.AwaitUntil("verify if service IP is discoverable", func() (interface{}, error) {
+			stdout, _, err := f.ExecWithOptions(framework.ExecOptions{
+				Command:       cmd,
+				Namespace:     f.Namespace,
+				PodName:       targetPod.Items[0].Name,
+				ContainerName: targetPod.Items[0].Spec.Containers[0].Name,
+				CaptureStdout: true,
+				CaptureStderr: true,
+			}, srcCluster)
+			if err != nil {
+				return nil, err
+			}
+
+			return stdout, nil
+		}, func(result interface{}) (bool, string, error) {
+			var doesContain bool
+			for _, serviceIp := range serviceIPList {
+				doesContain = strings.Contains(result.(string), serviceIp)
+				if doesContain {
+					serviceIPMap[serviceIp]++
+				}
+			}
+			By(fmt.Sprintf("Validating that dig result %s %q", op, result))
+			return true, "", nil
```

Comment:
I think we should record the returned IPs in an array so we can report it on failure:

 `retIPs := make([]string, 0)`
 
```suggestion
			for _, serviceIP := range serviceIPList {
				if strings.Contains(result.(string), serviceIP) {
					serviceIPMap[serviceIP]++
					retIPs = append(retIPs, serviceIP)
					break
				}
			}
			
			return true, "", nil
```